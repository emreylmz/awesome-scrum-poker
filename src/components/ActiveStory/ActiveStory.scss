@import "../../styles/colors";
@import "../../styles/metrics";

.active-story {
  display: flex;
  flex: 1;
  align-items: center;
  flex-direction: column;
  border: $default-border-width solid $primary-color;
  padding: $default-margin;
  border-radius: $default-border-radius;
  margin: $default-margin;

  &__points-container {
    display: flex;
    flex-wrap: wrap;
    .button {
      display: flex;
      flex: 1;
      align-items: center;
      justify-items: center;
      text-align: center;
      margin: $default-margin;
      padding: $default-margin;
      border: $default-border-width solid $primary-color;
      border-radius: $default-border-radius;

      &.selected {
        border-color: $secondary-color;
      }
    }
  }
}
