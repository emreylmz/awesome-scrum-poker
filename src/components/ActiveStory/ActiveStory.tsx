import React from 'react';

interface Props {

}

interface State {

}

class ActiveStory extends React.PureComponent<Props, State> {
  render() {
    return (
      <div>Active Story</div>
    );
  }
}

export default ActiveStory;
