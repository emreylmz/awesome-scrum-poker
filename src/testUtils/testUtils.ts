export const emptyTestFunc = (): void => void 0
