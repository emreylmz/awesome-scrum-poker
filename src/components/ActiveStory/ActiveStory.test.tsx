import { shallow, ShallowWrapper } from 'enzyme'
import React from 'react'
import sinon from 'sinon'
import '../../testUtils/setupTests'
import { emptyTestFunc } from '../../testUtils/testUtils'
import { ActiveStory, ActiveStoryProps, PointButton } from './ActiveStory'

type DoneCallback = jest.DoneCallback

const getWrapper = (props?: ActiveStoryProps): ShallowWrapper => {
  return shallow(<ActiveStory selectPoint={emptyTestFunc} {...props} />)
}

describe('Active Story UI', (): void => {
  it('should render correctly', (done: DoneCallback): void => {
    const wrapper = getWrapper()
    expect(wrapper).toMatchSnapshot()
    done()
  })

  it('should render story points correctly', (done: DoneCallback): void => {
    const point = '13'
    const wrapper = getWrapper({ selectedPoint: point, selectPoint: emptyTestFunc })
    const voteComponent = wrapper.find('div.active-story__selected-point')
    expect(voteComponent.text()).toEqual(`${point} Voted`)
    done()
  })

  it('should show selected story points', (done: DoneCallback): void => {
    const wrapper = getWrapper({ selectPoint: emptyTestFunc })
    let voteComponent = wrapper.find('div.active-story__selected-point')
    expect(voteComponent.text()).toEqual('Please Vote!!!')
    const point = '13'
    wrapper.setProps({
      selectedPoint: point
    })
    voteComponent = wrapper.find('div.active-story__selected-point')
    expect(voteComponent.text()).toEqual(`${point} Voted`)
    done()
  })
})

describe('Active Story Logic', (): void => {
  it('should call onSelect method when a point selected', (done: DoneCallback): void => {
    const testPoint = '13'
    const testFunc = sinon.spy()
    const wrapper = getWrapper({ selectPoint: testFunc })
    const pointButtonWrapper = wrapper.find(PointButton).at(0)
    pointButtonWrapper.invoke('onClick')(testPoint)

    expect(testFunc.calledOnce).toBeTruthy()
    expect(testFunc.args[0][0]).toEqual(testPoint)
    done()
  })
})
