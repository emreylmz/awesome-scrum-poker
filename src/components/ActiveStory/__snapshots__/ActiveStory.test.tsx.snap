// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active Story UI should render correctly 1`] = `ShallowWrapper {}`;
